src/js/angular/import/services/import-resource-tree.service.js

```diff
@@ -0,0 +1,193 @@
+    /**
+     * Convert list with {@link ImportResource} to a list of {@link ImportResourceTreeElement}.
+     * @param {ImportResource[]} importResources list of {@link ImportResource}
+     * @param {boolean} isUserImport - true if resource is from user.
+     * @return {ImportResourceTreeElement} the root of the import user resource tree.
+     */
+    static toImportResourceTree(importResources, isUserImport) {
+        const root = new ImportResourceTreeElement();
+        importResources.forEach((resource) => {
+            ImportResourceTreeService.addResourceToTree(root, resource, isUserImport);
+        });
+        return root;
+    }
+
+    /**
+     * Adds the <code>resource</code> into the tree with root <code>root</code>.
+     * @param {ImportResourceTreeElement} root
+     * @param {ImportResource} resource
+     * @param {boolean} isUserImport - true if resource is from user.
+     *
+     * @return {ImportResourceTreeElement} added resource tree element.
+     */
+    static addResourceToTree(root, resource, isUserImport) {
+        const path = ImportResourceTreeService.getPath(resource);
+        let directoryPath;
+        if (resource.isDirectory()) {
+            directoryPath = path;
+        } else {
+            directoryPath = path.slice(0, path.length - 1);
+        }
+
+        const resourceParent = ImportResourceTreeService.getOrCreateParent(root, directoryPath);
+
+        if (resource.isFile()) {
+            const importServerResource = new ImportResourceTreeElement();
+            importServerResource.parent = resourceParent;
+            importServerResource.importResource = resource;
+            importServerResource.path = path.join('/');
+            importServerResource.name = path[path.length - 1];
+            resourceParent.addResource(importServerResource);
+            ImportResourceTreeService.setIcon(importServerResource, isUserImport);
+            return importServerResource;
+        } else {
+            resourceParent.importResource = resource;
+            ImportResourceTreeService.setIcon(resourceParent, isUserImport);
+            return resourceParent;
+        }
+    }
+
+    static setIcon(importResourceElement, isUserImport) {
+        if (isUserImport) {
+            importResourceElement.iconClass = userImportResourceTypeToIconMapping.get(importResourceElement.importResource.type);
+        } else {
+            importResourceElement.iconClass = serverImportResourceTypeToIconMapping.get(importResourceElement.importResource.type);
+        }
+    }
+
+    /**
+     * Creates directory path (if no created yet) and returns the inner parent .
+     * @param {ImportResourceTreeElement} importServerResource - the root parent directory.
+     * @param {string[]} directoryPath - the list with directory names that have to be created.
+     *
+     * @return {ImportResourceTreeElement} the inner parent.
+     */
+    static getOrCreateParent(importServerResource, directoryPath) {
```

Comment:
Why `importServerResource`? Isn't this supposed to be generic component?

Reply:
Renamed to `importResourceElement`